File: core-service/src/main/java/com/sandjelkovic/dispatchd/domain/data/repository/UpdateJobRepository.java
package com.sandjelkovic.dispatchd.domain.data.repository;


import com.sandjelkovic.dispatchd.domain.data.entity.UpdateJob;
import org.springframework.data.repository.CrudRepository;

import java.util.Optional;

public interface UpdateJobRepository extends CrudRepository<UpdateJob, Long> {
	Optional<UpdateJob> findFirstByOrderByFinishTimeDesc();

	Optional<UpdateJob> findFirstBySuccessOrderByFinishTimeDesc(boolean success);
}

File: content-service/src/main/kotlin/com/sandjelkovic/dispatchd/contentservice/data/repository/UpdateJobRepository.kt
package com.sandjelkovic.dispatchd.contentservice.data.repository

import com.sandjelkovic.dispatchd.contentservice.data.entity.UpdateJob
import org.springframework.data.repository.CrudRepository
import java.util.*

/**
 * @author sandjelkovic
 * @date 3.3.18.
 */

interface UpdateJobRepository : CrudRepository<UpdateJob, Long> {
    fun findFirstByOrderByFinishTimeDesc(): Optional<UpdateJob>

    fun findFirstBySuccessOrderByFinishTimeDesc(success: Boolean): Optional<UpdateJob>
}
